### PR TITLE
add SNP annotations

### PR DESCRIPTION
**Pull request by cdebourcy**
- Modify DAG to generate a SNP annotation file (using the SNPs on which the tree was based + an annotated reference genome to compare against)
- Make the file available for download

### CODE REVIEW

File: app/assets/src/components/ui/controls/buttons/DownloadButton.jsx [hunk 1]
```diff
@@ -15,6 +19,7 @@ const DownloadButton = ({ disabled, onClick }) => {
 };
 
 DownloadButton.propTypes = {
+  what: PropTypes.string,
```
Comment (tfrcarvalho) [inline app/assets/src/components/ui/controls/buttons/DownloadButton.jsx]: Can you check with Jenn on this ? 
If we need to set the label, I would probably would go with the DropdownDownloadButton to keep consistency of the UI.

File: app/models/phylo_tree.rb [hunk 2]
```diff
@@ -61,12 +64,29 @@ def job_command
...
+      top_taxid_by_run_id = {}
+      species_counts.each do |sc|
+        top_taxid_by_run_id[sc.pipeline_run_id] ||= sc.tax_id
+      end
+      reference_taxids = top_taxid_by_run_id.values
+    else
+      reference_taxids = [taxid]
+    end
+    # Retrieve superkigdom name for idseq-dag
+    superkingdom_name = TaxonLineage.where(taxid: 573).last.superkingdom_name
```
Comment (tfrcarvalho) [inline app/models/phylo_tree.rb]: What is 573 taxid? Can we add it to a constant for readability?

File: app/assets/src/components/ui/controls/buttons/DownloadButton.jsx (hunk 1 above)
Comment (cdebourcy) [reply app/assets/src/components/ui/controls/buttons/DownloadButton.jsx]: ok will go with DropdownDownloadButton

File: app/models/phylo_tree.rb (hunk 2 above)
Comment (cdebourcy) [reply app/models/phylo_tree.rb]: oops that's a glitch, it should be `where(taxid: taxid)`, good catch

File: app/controllers/phylo_trees_controller.rb [hunk 3]
```diff
@@ -97,6 +97,12 @@ def retry
     end
   end
 
+  def download_snps
+    snp_file = Tempfile.new
+    Open3.capture3("aws", "s3", "cp", @phylo_tree.s3_outputs["SNP_annotations"], snp_file.path)
+    send_file snp_file.path, filename: "#{@phylo_tree.name.downcase.gsub(/\W/, '-')}__SNP-annotations.txt"
```
Comment (jshoe) [inline app/controllers/phylo_trees_controller.rb]: for Open3.capture3 could you do something `unless status.success? LogUtil.log_err_and_airbrake("downloading #{@phylo_tree.s3_outputs["SNP_annotations"]} failed")`

Comment (cdebourcy) [reply app/controllers/phylo_trees_controller.rb]: sure

**Pull request merged**